userguide/tutorials/payment_plugin.adoc

```diff
@@ -463,147 +475,148 @@ public PaymentTransactionInfoPlugin purchasePayment(UUID kbAccountId, UUID kbPay
+** *PaymentPluginStatus* - Set to `PaymentPluginStatus.CANCELED` since the gateway was not contacted as the plugin was unable to retrieve the `mandateId`.
 
-** *gatewayError* - Set to `null` since there is no error
+** *gatewayError* - Set to `null` since there is no error.
 
-** *gatewayErrorCode* - set to `null` since there is no error
+** *gatewayErrorCode* - set to `null` since there is no error.
 
-** *firstPaymentReferenceId* - Set to `null` since the payment was not processed
+** *firstPaymentReferenceId* - Set to `null` since the payment was not processed.
 
-** *effectiveDate* - Set to `null` since the payment was not processed
+** *effectiveDate* - Set to `null` since the payment was not processed.
 
-** *properties* - Set to `null` since the payment was not processed
+** *properties* - Set to `null` since the payment was not processed.
 
+[[getPaymentInfo]]
 ==== Step 7 - Implementing GoCardlessPaymentPluginApi#getPaymentInfo
 
-As explained earlier, the Kill Bill Janitor system attempts to fix *PENDING* and *UNKNOWN* states by  polling the plugin via `PaymentPluginApi#getPaymentInfo` method. So, this method can be implemented as follows (See https://github.com/killbill/killbill-gocardless-example-plugin/blob/9522498ecde5849c940574c598ceb5ce088d32a7/src/main/java/org/killbill/billing/plugin/gocardless/GoCardlessPaymentPluginApi.java#L211[_GoCardlessPaymentPluginApi#getPaymentInfo_]):
+The `getPaymentInfo` method is used to retrieve payment information. It is very important to implement this method properly since it is used by the https://docs.killbill.io/latest/userguide_payment.html#_janitor[__ Kill Bill Janitor__]. The Janitor attempts to fix *PENDING* and *UNKNOWN* transaction states. It queries the plugin via the `getPaymentInfo` method. It then updates the Kill Bill database based on the data in the `PaymentTransactionInfoPlugin` object returned by the `getPaymentInfo` method. So, if the plugin subsequently returns a *PROCESSED* status, the Janitor updates the internal payment state accordingly. In addition, the Janitor also updates other fields in the `PaymentTransactionInfoPlugin` object like processed amount (via `PaymentTransactionInfoPlugin#getAmount`), error information (via `PaymentTransactionInfoPlugin#getGatewayError`/`PaymentTransactionInfoPlugin#getGatewayErrorCode`), etc. in the Kill Bill database. 
+
+The Janitor matches the internal transactions against plugin transactions via the *transaction id*, so make sure `PaymentTransactionInfoPlugin#getKbTransactionPaymentId` is correctly implemented.
+
+In addition to the Janitor, the `getPaymentInfo` is also invoked whenever a https://killbill.github.io/slate/#payment-retrieve-a-payment-using-paymentid[payment is retrieved] with `withPluginInfo=true` which in turn results in triggering the https://docs.killbill.io/latest/userguide_payment.html#_on_the_fly_janitor[__on the fly janitor__] to fix the payment state before returning the payment information. 
```

Comment:
```suggestion
In addition to the Janitor, the `getPaymentInfo` is also invoked whenever a https://killbill.github.io/slate/#payment-retrieve-a-payment-using-paymentid[payment is retrieved] with `withPluginInfo=true`, which in turn results in triggering the https://docs.killbill.io/latest/userguide_payment.html#_on_the_fly_janitor[on the fly janitor] to fix the payment state before returning the payment information. 
```

Reply:
^^^Does "on the fly janitor" need to be capitalized?

Reply:
Don't think so, I've changed it to "on-the-fly Janitor" based on https://docs.killbill.io/latest/userguide_payment.html#_on_the_fly_janitor